internal/jimm/cloud.go

```diff
@@ -600,24 +600,27 @@ func (j *JIMM) UpdateCloud(ctx context.Context, user *openfga.User, ct names.Clo
 	// do this in a transaction so that the local view cannot finish in
 	// an inconsistent state.
 	err = j.Database.Transaction(func(db *db.Database) error {
+		var transactionCloud dbmodel.Cloud
+		transactionCloud.SetTag(ct)
 
-		var c dbmodel.Cloud
-		c.SetTag(ct)
-		if err := db.GetCloud(ctx, &c); err != nil {
+		if err := db.GetCloud(ctx, &transactionCloud); err != nil {
 			return err
 		}
-		c.FromJujuCloud(cloud)
-		for i := range c.Regions {
-			if len(c.Regions[i].Controllers) == 0 {
-				for _, ctl := range controllers {
-					c.Regions[i].Controllers = append(c.Regions[i].Controllers, dbmodel.CloudRegionControllerPriority{
-						Controller: ctl,
-						Priority:   dbmodel.CloudRegionControllerPrioritySupported,
-					})
-				}
-			}
+		transactionCloud.FromJujuCloud(cloud)
+
+		// Set the region controller priorities again for this transactions dbmodel.Cloud.
+		for i := range transactionCloud.Regions {
```

Comment:
Yeah I'm not sure either really, it's not clear why the == 0 check was there. And on the transaction, I think it's because we're retrieving the cloud first and in theory the cloud could change between the time you first retrieve it and then set it. So I think a transaction makes sense here.

Reply:
the thing is that you are not using the `transactionCloud`, you are just overriding the regions field, so you can do that w/o using a transaction. Right?

Reply:
Ahhh, I see what you mean. Perhaps it should be getting the cloud again, good catch.